ml-algorithms/src/test/java/org/opensearch/ml/engine/httpclient/MLHttpClientFactoryTests.java

```diff
@@ -6,99 +6,145 @@
 package org.opensearch.ml.engine.httpclient;
 
 import static org.junit.Assert.assertNotNull;
+import static software.amazon.awssdk.http.SdkHttpMethod.POST;
 
-import java.net.UnknownHostException;
+import java.net.URI;
+import java.nio.charset.Charset;
 
-import org.apache.http.HttpHost;
-import org.apache.http.impl.client.CloseableHttpClient;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.ExpectedException;
 
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.http.SdkHttpFullRequest;
+import software.amazon.awssdk.http.async.SdkAsyncHttpClient;
+
 public class MLHttpClientFactoryTests {
 
     @Rule
     public ExpectedException expectedException = ExpectedException.none();
 
     @Test
-    public void test_getCloseableHttpClient_success() {
-        CloseableHttpClient client = MLHttpClientFactory.getCloseableHttpClient();
+    public void test_getSdkAsyncHttpClient_success() {
+        SdkAsyncHttpClient client = MLHttpClientFactory.getAsyncHttpClient();
         assertNotNull(client);
     }
 
     @Test
-    public void test_validateIp_validIp_noException() throws UnknownHostException {
-        MLHttpClientFactory.validateIp("api.openai.com");
+    public void test_validateIp_validIp_noException() {
+        RequestBody requestBody = RequestBody.fromString("hello world", Charset.defaultCharset());
+        SdkHttpFullRequest request = SdkHttpFullRequest
+            .builder()
+            .method(POST)
+            .uri(URI.create("https://api.openai.com"))
+            .contentStreamProvider(requestBody.contentStreamProvider())
+            .build();
     }
 
     @Test
-    public void test_validateIp_invalidIp_throwException() throws UnknownHostException {
-        expectedException.expect(UnknownHostException.class);
-        MLHttpClientFactory.validateIp("www.zaniu.com");
-    }
-
-    @Test
-    public void test_validateIp_privateIp_throwException() throws UnknownHostException {
-        expectedException.expect(IllegalArgumentException.class);
-        MLHttpClientFactory.validateIp("localhost");
-    }
-
-    @Test
-    public void test_validateIp_rarePrivateIp_throwException() throws UnknownHostException {
+    public void test_validateIp_rarePrivateIp_throwException() {
+        RequestBody requestBody = RequestBody.fromString("hello world", Charset.defaultCharset());
         try {
-            MLHttpClientFactory.validateIp("0254.020.00.01");
-        } catch (IllegalArgumentException e) {
+            SdkHttpFullRequest request = SdkHttpFullRequest
+                .builder()
+                .method(POST)
+                .uri(URI.create("http://0177.1/v1/completions"))
+                .contentStreamProvider(requestBody.contentStreamProvider())
+                .build();
+        } catch (NullPointerException e) {
```

Comment:
Will this throw 500 error ? Seems not correct, as it caused by wrong input

Reply:
Changed the validate code a little bit to make it more compatible with existing UTs